app/scripts/lib/experiment.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'lib/url',
+  'lib/experiments/open-gmail',
+  'lib/experiments/mailcheck',
+  'lib/experiments/coppa-view',
+  'lib/experiments/sync-checkbox'
+], function (_, Url, OpenGmailExperiment, MailcheckExperiment, CoppaExperiment, SyncCheckbox) {
+  'use strict';
+
+  var UA_OVERRIDE = 'FxATester';
+  var CHOOSE_ABLE_EXPERIMENT = 'chooseAbExperiment';
+  var FORCE_EXPERIMENT_PARAM = 'forceExperiment';
+  // all available experiments that must be independent for A/B testing purposes.
+  var DEFAULT_EXPERIMENTS = {
+    'openGmail': OpenGmailExperiment,
+    'mailcheck': MailcheckExperiment,
+    'coppaView': CoppaExperiment,
+    'syncCheckbox': SyncCheckbox
+  };
+
+  function ExperimentInterface (options) {
+    if (! options || ! options.window || ! options.able || ! options.metrics || ! options.user || ! options.notifications) {
+      this.initialized = false;
+      return;
+    }
+
+    this.window = options.window;
+    var agent = this.window.navigator.userAgent;
+    this.forceExperiment = Url.searchParam(FORCE_EXPERIMENT_PARAM, this.window.location.search);
+
+    // if this is running in functional test mode then we do not want any unpredictable experiments
+    if (agent.indexOf(UA_OVERRIDE) >= 0 && ! this.forceExperiment) {
```

Comment:
Can you make a quick doc about the functional tests set a special user agent string, otherwise this comment lacks a little context and I'll forget what's going on 6 months from now. As an aside, is that special UA string set universally, or only for a certain set of tests?